src/factom-event-emitter.js

```diff
@@ -186,31 +268,65 @@ class FactomEventEmitter extends EventEmitter {
                 this._lastBlockHeightProcessed = block.height;
                 this._handleDirectoryBlock(block);
             }
+            if (this._factoidAddressPendingTransactionSubscriptions.size > 0) {
+                this._checkPendingTransactions();
+            }
         } catch (err) {
             this.emit('error', err);
         }
     }
 
+    // Check pending address subscriptions for any pending transactions
+    async _checkPendingTransactions() {
+        for (const fctAddress of this._factoidAddressPendingTransactionSubscriptions.keys()) {
+            const pendingTransactions = await this.getPendingTransactions(fctAddress);
+
+            if (pendingTransactions.length > 0) {
+                this._handlePendingTransactions(fctAddress, pendingTransactions);
+            }
+        }
+    }
+
+    /* Trigger pending transaction emitter if there are new pending transactions
+     * and update subscribed transactions
+     */
+    _handlePendingTransactions(fctAddress, pendingTransactions) {
+        const subscribedTransactionIds = this._factoidAddressPendingTransactionSubscriptions.get(
+            fctAddress
+        );
+
+        for (const pendingTransaction of pendingTransactions) {
+            if (!subscribedTransactionIds.has(pendingTransaction.id)) {
```

Comment:
@afenrir mentioned it, and I think he is right, it seems that this `id` should actually be `transactionid`, isn't it?

Could you also write a test that would test this dedupe logic, checking that a pending transaction is never re-emitted?